descwl_coadd/coadd.py

```diff
@@ -1,1236 +1,639 @@
+        The random number generator for making noise images
+    remove_poisson: bool
+        If True, remove the poisson noise from the variance
+        estimate.
     loglevel : str, optional
         The logging level. Default is 'info'.
 
-    Attributes
-    ----------
-    interp : str
-        The kind of interpolation used for the coadd. Currently always lanczos3.
-    coadd_psf_wcs : DM stack sky WCS object
-        The WCS for the PSF exposure.
-    coadd_psf_exp : DM stack exposure
-        The coadded PSF.
-    coadd_exp : DM stack exposure
-        The coadded image.
-    coadd_noise_exp : DM stack exposure
-        The coadded noise image.
+    Returns
+    -------
+    ExposureF for coadd
     """
-    def __init__(self, *,
-                 exps,
-                 noise_exps,
-                 psf_exps,
-                 coadd_wcs,
-                 coadd_dims,
-                 psf_dims,
-                 loglevel='info'):
-
-        import galsim
-
-        self.log = lsst.log.getLogger("CoaddObs")
-        self.log.setLevel(getattr(lsst.log, loglevel.upper()))
-        self.loglevel = loglevel
-
-        self.exps = exps
-        self.psf_exps = psf_exps
-        self.noise_exps = noise_exps
-        self.galsim_wcs = coadd_wcs
-        self.coadd_wcs = make_stack_wcs(coadd_wcs)
-        self.coadd_dims = coadd_dims
-        self.psf_dims = psf_dims
-
-        ceny, cenx = (np.array(coadd_dims)-1)/2
-        image_pos = galsim.PositionD(x=cenx, y=ceny)
-        jac = coadd_wcs.local(image_pos=image_pos)
-
-        self.coadd_psf_wcs = make_stack_psf_wcs(
-            dims=psf_dims,
-            offset=galsim.PositionD(x=0, y=0),
-            jac=jac,
-            world_origin=coadd_wcs.center,
-        )
-
-        self.interp = 'lanczos3'
-
-        self._make_coadds()
-        self._finish_init()
 
-    def show(self):
-        """show the output coadd in DS9"""
-        self.log.info('showing coadd in ds9')
-        vis.show_image_and_mask(self.coadd_exp)
-        # this will block
-        vis.show_images(
-            [
-                self.image,
-                self.coadd_exp.mask.array,
-                self.noise,
-                self.coadd_noise_exp.mask.array,
-                self.coadd_psf_exp.image.array,
-                # self.weight,
-            ],
-        )
+    logger = make_logger('coadd', loglevel)
 
-    def _make_coadds(self):
-        """
-        make warps and coadds for images and noise fields
-        """
-        image_data = self._make_warps(
-            exps=self.exps,
-            dims=self.coadd_dims,
-            wcs=self.coadd_wcs,
-            dopsf=True,
-        )
-
-        psf_data = self._make_warps(
-            exps=self.psf_exps,
-            dims=self.psf_dims,
-            wcs=self.coadd_psf_wcs,
-            dopsf=False,
-        )
+    coadd_psf_bbox = geom.Box2I(
+        geom.Point2I(0, 0),
+        geom.Point2I(psf_dims[0]-1, psf_dims[1]-1),
+    )
```

Comment:
Are bounding boxes supposed to start at zero or should this be centered on a stamp at the central pixel of the coadd?

Reply:
need to double check

Reply:
I think zero is correct